crates/ruff_linter/src/rules/eradicate/detection.rs

```diff
@@ -48,65 +53,36 @@ pub(crate) fn comment_contains_code(line: &str, task_tags: &[String]) -> bool {
         return false;
     }
 
-    // Ignore non-comment related hashes (e.g., "# Issue #999").
-    if HASH_NUMBER.is_match(line) {
-        return false;
-    }
-
     // Ignore whitelisted comments.
     if ALLOWLIST_REGEX.is_match(line) {
         return false;
     }
 
-    if CODING_COMMENT_REGEX.is_match(line) {
-        return false;
-    }
-
-    // Check that this is possibly code.
-    if CODE_INDICATORS.iter().all(|symbol| !line.contains(symbol)) {
+    // Ignore non-comment related hashes (e.g., "# Issue #999").
+    if HASH_NUMBER.is_match(line) {
         return false;
     }
 
-    if multiline_case(line) {
-        return true;
-    }
-
-    if CODE_KEYWORDS.iter().any(|symbol| symbol.is_match(line)) {
+    // If the comment made it this far, and ends in a continuation, assume it's code.
+    if line.ends_with('\\') {
         return true;
     }
 
-    let line = PRINT_RETURN_REGEX.replace_all(line, "");
-
-    if PARTIAL_DICTIONARY_REGEX.is_match(&line) {
+    // If the comment matches any of the specified positive cases, assume it's code.
+    if POSITIVE_CASES.is_match(line) {
```

Comment:
Interestingly, combining `HASH_NUMBER` and `ALLOWLIST_REGEX` into a single regex ended up _hurting_ performance, so I left those as two separate passes.